integration/jnifuse/fs/src/main/java/alluxio/jnifuse/LibFuse.java

```diff
@@ -13,52 +13,56 @@
 
 import alluxio.jnifuse.utils.LibfuseVersion;
 import alluxio.jnifuse.utils.NativeLibraryLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.util.concurrent.atomic.AtomicReference;
 
+/**
+ * Load the libfuse library.
+ */
 public class LibFuse {
+  private static final Logger LOG = LoggerFactory.getLogger(LibFuse.class);
 
-  private enum LibraryState {
-    NOT_LOADED,
-    LOADING,
-    LOADED
+  /**
+   * The libfuse load strategy.
+   */
+  public enum LibfuseLoadStrategy {
+    LOAD_FUSE2_ONLY,
+    LOAD_FUSE3_ONLY,
+    FUSE3_THEN_FUSE2,
   }
 
-  private static AtomicReference<LibraryState> libraryLoaded =
-      new AtomicReference<>(LibraryState.NOT_LOADED);
```

Comment:
Does not need the complicated concurrency logics, since the library will be loaded only once